<launch>

  <include file="$(find state_manager)/launch/stateManager.launch" />
  <!-- ============================= Namespace ============================= -->

  <arg name="namespace" default="pandora_vision/hole_detector" />

  <!-- ========================= Synchronizer node ========================= -->

  <node
    name="synchronizer_node"
    type="synchronizer_node"
    pkg="pandora_vision_hole"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole)/cfg/synchronizer_node_topics.yaml"
              command="load" />
  </node>

  <!-- ============================ Depth node ============================= -->

  <node
    name="depth_node"
    type="depth_node"
    pkg="pandora_vision_hole"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole)/cfg/depth_node_topics.yaml"
              command="load" />
  </node>

  <!-- ============================= RGB node ============================== -->

  <node
    name="rgb_node"
    type="rgb_node"
    pkg="pandora_vision_hole"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole)/cfg/rgb_node_topics.yaml"
              command="load" />
  </node>

  <!-- ========================= Hole Fusion node ========================== -->

  <node
    name="hole_fusion_node"
    type="hole_fusion_node"
    pkg="pandora_vision_hole"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole)/cfg/hole_fusion_node_topics.yaml"
              command="load" />
  </node>

  <rosparam file="$(find pandora_vision_support)/config/camera_benchmark_params.yaml" 
    command="load"/>

  <param name="subscriberTopic" value="/vision/holes_direction_alert" />
  <param name="subscriberMessagePackage" value="pandora_vision_msgs" />
  <param name="subscriberMessageType" value="HolesDirectionsVectorMsg"/>

  <param name="publisherTopic" value="/camera/depth_registered/points" />
  <param name="publisherMessagePackage" value="sensor_msgs" />
  <param name="publisherMessageType" value="PointCloud2"/>
  
  <param name="algorithm" value="Hole"/>
  <remap from="/camera/depth_registered/points" to="/kinect/depth_registered/points" />

  <param name="dataset_camera" value="kinect"/>
  <param name="dataset_path" value="/test/benchmark/images/bags_pcl"/>
  
  <test test-name="HoleBenchmarkTest" pkg="pandora_vision_hole"
    type="hole_benchmark_test.py" time-limit="1000.0"/>
  
</launch>
